design_docs/generalized_horizontal_interpolation.md

```diff
@@ -0,0 +1,95 @@
+<h1> Title: Generalized Nearest-neighbor and First-order Interpolation in MPAS-Analysis <br>
+Xylar Asay-Davis <br>
+date: 01-19-2017 <br>
+</h1>
+<h2> Summary </h2>
+Many types of analysis in MPAS will require fields that are interpolated from MPAS grids to arbitrary points.  Currently, nearest-neighbor interpolation is supported in MPAS-Analysis, but only for points on a uniformly spaced lat-lon grid.  Building on the existing approach, this design document describes methods for performing both nearest-neighbor and first-order (linear or similar) interpolation from MPAS grids to arbitrary points.
+
+<h1> Requirements </h1>
+<h2>Requirement: Support for arbitrary output interpolation points
+Date last modified: 2017/01/19
+Contributors: Xylar Asay-Davis
+</h2>
+
+The calling code should be able to supply any desired interpolation points, not just a regular latitude-longitude grid.
+
+<h2>Requirement: Cartesian input mesh
+Date last modified: 2017/01/19
+Contributors: Xylar Asay-Davis
+</h2>
+
+The source mesh should be described in Cartesian, rather than lon-lat, space.  This is more consistent with how MPAS meshes are constructed (e.g. they have Voronoi cells only in Cartesian coordinates, not in lon-lat coordinates) and also is more conducive to supporting both spherical and planar meshes.
+
+<h2>Requirement: First-order interpolation
+Date last modified: 2017/01/19
+Contributors: Xylar Asay-Davis</h2>
+
+The option to interpolate smoothly (e.g. linearly or with barycentric coordinates) between cell-centered values should be added.  The calling code should easily be able to select either nearest-neighbor or first-order interpolation with a flag as part of the initialization step is required and from then on should not need to keep track of which method is being used.
```

Comment:
> The calling code should easily be able to select either nearest-neighbor or first-order interpolation with a flag as part of the initialization step is required and from then on should not need to keep track of which method is being used.

to 

The calling code should easily be able to select either nearest-neighbor or first-order interpolation via a flag at initialization.  Subsequently routines use this initial choice of nearest-neighbor or first-order interpolation.